### ISSUE TITLE
Colorscript bug

### ISSUE THREAD

**Issue by 0xN1nja** [gh0stzk/dotfiles#256]
So, this issue has been happening since I added your latest colorscript to my config. The colorscript is not executing `zwaves`. It is executable when run manually:

![image](https://github.com/gh0stzk/dotfiles/assets/71812498/34743b6c-2632-4eaf-8420-eb7e1cc8ecd8)

Even colorscript can execute it:

![image](https://github.com/gh0stzk/dotfiles/assets/71812498/acb1fedc-642f-4d56-be59-d72dcc5f5b6a)

But the `colorscript -r` command never executes it. I've tried running `colorscript -r` over 50 times, but not a single time it executed `zwaves`. This is very weird, I don't know why it's happening.

`zwaves` even gets listed in the output of `colorscript -l`:

![image](https://github.com/gh0stzk/dotfiles/assets/71812498/33537bab-167f-4490-ac12-a4b972b8e8e7)

**Comment by AzhamProdLive** [comment]
![image](https://github.com/gh0stzk/dotfiles/assets/73955243/1177aec1-9d9e-4d73-8f98-355450891eb1)
Can reproduce, error is here

**Comment by 0xN1nja** [comment]
@AzhamProdLive wrong command. `-r` flag is used to run a random colorscript.

**Comment by AzhamProdLive** [comment]
> @AzhamProdLive wrong command. `-r` flag is used to run a random colorscript.

Yeah I noticed after I sent it, you just happen to have bad luck, it works on my machine

![image](https://github.com/gh0stzk/dotfiles/assets/73955243/e30ce238-b2b4-48c6-9763-32015217e8eb)

**Comment by 0xN1nja** [comment]
That's weird :/

**Comment by AzhamProdLive** [comment]
Yes, I do have the latest

**Comment by AzhamProdLive** [comment]
I get all of them, from those I still have (so only 8), can't reproduce

**Comment by gh0stzk** [comment]
Hello, is a legit bug as seen here https://gitlab.com/dwt1/shell-color-scripts/-/commit/32e2feefcdd01dcf348b22749ffd9875f8438b25 But the version that the dotfiles use is this https://github.com/charitarthchugh/shell-color-scripts. Before, I used the one from Derek Taylor Distro Tube but it kept getting bloated and I changed to the recent one.

I can't reproduce the bug either since it appears, in my case it took me 9 attempts.

![Shot-2024-06-02-124558](https://github.com/gh0stzk/dotfiles/assets/67278339/e722f442-50fb-4d6e-8465-6f4205416ed1)

2 new themes are coming, and a bluetooth manager in rofi with his respective polybar module.

**Comment by 0xN1nja** [comment]
I finally found the bug. It's on line 26 of colorscript. We are starting the `random_index` from 1; not 0, which is why the "nth" (last) script is not being executed.

Whenever there were 2 scripts in the colorscripts directory, only 1 was being executed. When there were 3 scripts in the colorscripts directory, only 2 were being executed randomly. And when there were 4 scripts, only 3 were being executed randomly.

Hence, n-1 scripts were being executed. The "nth" script (or the last script) was never being executed.

So, I changed the `random_index` from:

```bash
declare -i random_index=$RANDOM%$length_colorscripts
```

to:

```bash
declare -i random_index=$((RANDOM % length_colorscripts + 1))
```

It took me a lot of time to figure this out, but now it's working. I don't know how it was working for you all, maybe you were missing something.
